content/articles/python-metaclasses-for-metaprogramming/index.md

```diff
@@ -0,0 +1,233 @@
+
+```python
+number = 11  
+print("The list is classified as follows::", type(number))  
+   
+list = [4, 6, 9, 10]  
+print("The list is classified as follows::", type(list))  
+   
+name = "Dickson Gitau"  
+print("The list is classified as follows::", type(name))
+```
+
+#### Output
+```python
+The list is classified as follows:: <class 'int'>
+The list is classified as follows:: <class 'list'>
+The list is classified as follows:: <class 'str'>
+```
+
+Python's class system is in charge of defining all of the language's types. The above example uses class `int,` `list,` or `str`, unlike other basic data types that include `int,` `char,` and `float` as found in C and Java.
+
+New data types may be generated using user-defined classes. We will use the `Town` class as a new object in an example.
+
+```Python
+class Town:  
+    pass  
+Town_object = Town()  
+print("Sort of Town_object is:", type(Town_object))
+```
+
+#### Output
+```python
+Sort of Town_object is: <class '__main__.Town'>
+```
+
+In Python, a class is also an object, and as such, it is an instance of Metaclass, just like other objects. Metaclasses are a unique form of class that is in charge of constructing classes and Class objects. If a person searches for the type of the `City` class, for example, they will discover that it is `type.`
+
+```Python
+class Town:  
+    pass   
+print("Sort of Town class is:", type(Town)) 
+```
+
+#### Output
+```python
+Sort of Town class is: <class 'type'>
+```
+
+We can alter the classes since they are also objects. For example, user-created properties and methods may be added or removed from the class.
+
+```Python
+class Town:  
+    pass  
+Town.b = 70
+Town.foo = lambda persona: print('Dusk')
+enjoyerobject = Town()  
+   
+print(enjoyerobject.b)  
+enjoyerobject.foo() 
+```
+
+#### Output
+
+```Python
+70
+Dusk
+```
+
+We may proceed by starting with a class that does not utilize any methods or variables from the class. Then, the method variables and class methods will be specified in the following steps. In the end, we will create the final object.
+
+### Reasons to use metaclasses instead of functions
+Although we have equally useful functions, we need metaclasses for the following reasons.
+- The aim of metaclasses is precise. At the point when you read `UpperAttrMetaclass(type)`, you know what will follow
+- You can utilize OOP. Metaclasses may inherit the parent method.
+- A class's subclasses are instances of its metaclass if the metaclass-class is given, but not the metaclass function.
+- You may improve the organization of your code. Using metaclasses for a simple example like this is a waste of time. It is usually reserved for a more difficult task. Making several methods and grouping them into a single class makes the code more readable.
+- It is possible to hook on `new`, `init`, and `call` hooks. 

```

Comment:
@Dick7603  These points are great but you need to clarify them. Provide some in-depth explanation for the benefit of readers.